sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/BulkWriter.scala

```diff
@@ -930,7 +930,24 @@ private class BulkWriter(container: CosmosAsyncContainer,
                   activeReadManyOperationsSnapshot,
                   numberOfIntervalsWithIdenticalActiveOperationSnapshots
                 )
+
+                if (numberOfIntervalsWithIdenticalActiveOperationSnapshots.get == 1L) {
+                  val activeWriteOperationsSnapshot = activeBulkWriteOperations.clone()
+                  activeWriteOperationsSnapshot.foreach(operation => {
+                    if (activeBulkWriteOperations.contains(operation)) {
```

Comment:
this we are re-emitting, so it is possible we are getting both response back. And when this happens, the `assume(itemOperationFound)` will throw errors?
```
    bulkOperationResponseFlux.subscribe(
      resp => {
        val isGettingRetried = new AtomicBoolean(false)
        try {
          val itemOperation = resp.getOperation
          val itemOperationFound = activeBulkWriteOperations.remove(itemOperation)
          assume(itemOperationFound) // can't find the item operation in list of active operations!
```

Reply:
Fixed

Reply:
Good catch!!!